packages/yarnpkg-core/sources/WorkspaceResolver.ts

```diff
@@ -55,7 +55,7 @@ export class WorkspaceResolver implements Resolver {
 
       version: workspace.manifest.version || `0.0.0`,
 
-      languageName: `unknown`,
+      languageName: workspace.manifest.languageName || opts.project.configuration.get(`defaultLanguageName`),
```

Comment:
I'll revert this part; as I mentioned in Discord, workspaces have no "language name" because they're everything at the same time.